### ISSUE TITLE
Stuck while loading playlists & tracks: "TypeError: data.tracks is null"

### ISSUE THREAD

**Issue by duracotton** [secuvera/SpotMyBackup#23]
I'm getting the javascript error above which results in a freezing of the loading process of tracks and playlists: 
"TypeError: data.tracks is null"

![unbenannt](https://user-images.githubusercontent.com/25416153/40929405-e57448f0-6824-11e8-9370-11c1447a4c15.PNG)

**Comment by MoraOgunro** [comment]
I'm having a similar issue. A lot of other Spotify backup tools are not working anymore either. I think Spotify changed something big recently.

**Comment by shape55** [comment]
Yes, when "Loading your playlists and tracks" I can't get past 376 playlists, but it's not even consistent. It loads a different amount every time and then hangs.

**Comment by shape55** [comment]
Yes, when "Loading your playlists and tracks" I can't get past 376 playlists, but it's not even consistent. It loads a different amount every time and then hangs.

In the Javascript Console it shows:
(index):640 Uncaught TypeError: Cannot read property 'href' of null

**Comment by mat-ale** [comment]
I think it would be good to put a try - catch [here](https://github.com/secuvera/SpotMyBackup/blob/master/index.html#L649), what do you think?

**Comment by RamonLM** [comment]
Same problem as #22.

> I think it would be good to put a try - catch here, what do you think?

Actually all the starred playlists logic should be removed as it's been deprecated by Spotify.

**Comment by kotos-net** [comment]
Export not working :(  My exporting stopped on  88 playlists 5900 tracks. Could you repair it. Thank you

**Comment by fgimian** [comment]
I can confirm the same issue on my end too 😢

**Comment by malicious50** [comment]
Same issue.. any of you found a suitable alternative to backup spotify playlists?

**Comment by focamonca** [comment]
Same issue!

**Comment by shape55** [comment]
@malicious50 - There is also Exportify but it's limited to 50 playlists for the Export All function:
https://github.com/watsonbox/exportify/issues/43 (see this issue for updated link).  I'm not sure if SpotMyBackup was intended to work with more than 50, however, as I didn't try it until a week ago, and the Spotify API has a 50 playlist return limit.

And also http://joellehman.com/playlist/ but this is even more limited at 20 playlists.

**Comment by RamonLM** [comment]
@shape55 Check https://gitlab.com/rwig/backify based on this project.

It has no playlist limitations, and added support for artists.

**Comment by shape55** [comment]
Hey, thanks a lot for working on this. Although it's not loading all my data I was able to export a 4.1 MB json file of most of the playlists using Firefox.

With both Firefox and Chrome I get the following: 

"4 artists (Obviously, this isn't correct)
371 playlists (I have 440 total)
55299 tracks

Finished loading, you now might want to export or import."

The Export button works in Firefox but doesn't respond in Chrome, though the Import button does open an import window.  In Safari, the Login With Spotify button doesn't respond.

Also, any chance you could add an option to export as csv like Exportify does so it saves all the fields as text?  This way the backup can be opened in a text editor to view all the data.

**Comment by RamonLM** [comment]
@shape55 Thank you for reporting this, will create an issue and look into it

**Comment by secuvera** [comment]
Removed the starred playlist since the api does not exist anymore (in source and on spotmybackup.com). Hope that helps.

**Comment by shape55** [comment]
@RamonLM - Hi, just tested this out to see if any fixes were made but i'm still having the same issues (though it's loading 373 playlists now rather than 371).

**Comment by shape55** [comment]
@secuvera – Though it's still not able to export in Chrome (Export button depresses but nothing downloads), after 3 attempts in Firefox I was able to get a json file to download.  But is there a way to view the playlist info (track, artist, album, etc.)?  When I opened it in Excel it's illegible b/c it's not formatted properly.  

My assumption w/ a backup was that the file would contain the full tracklists of each playlist and all corresponding fields the way Exportify does.  However, Exportify doesn't allow you to export more than 50 playlists using the "Export All" function.  Is it possible to add an option to download the backup as a csv file w/ all fields present?

**Comment by watsonbox** [comment]
> However, Exportify doesn't allow you to export more than 50 playlists using the "Export All" function.

It should do now 🙂 . More info on all the recent updates: https://watsonbox.github.io/posts/2020/12/02/exportify-refresh.html.

**Comment by shape55** [comment]
FYI, if you guys haven't tried [Pavel Komarov's fork](https://github.com/pavelkomarov/exportify) of watsonbox's Exportify, he solved the rate-limiting error in 2019.  So "Export All" works well now.  It can be accessed at [exportify.net](https://exportify.net).
